tests/python/gaia-ui-tests/gaiatest/gaia_test.py

```diff
@@ -1055,6 +1054,15 @@ def is_element_displayed(self, by, locator):
     def tearDown(self):
         if self.device.is_desktop_b2g and self.device.storage_path:
             shutil.rmtree(self.device.storage_path, ignore_errors=True)
+        if self.device.is_android_build:
+            file_items = self.device.file_manager.list_items('/data/local/webapps/')
+            dirs_to_keep = ['{14750155-5f3d-41ab-a527-f16adf4c3293}', 'marketplace-dev.allizom.org', 'marketplace.allizom.org', 'marketplace.firefox.com']
```

Comment:
I think one issue here is that some builds put the system apps into `/data/local/webapps`, so we'd still be deleting them. I really don't like the idea of hardcoding the directories to keep. Looking at `/data/local/webapps/webapps.json` is appears that we could identify the apps to delete using `"removable": true`. Could you try a patch that reads in that file, identifies the removable apps and deletes them from the filesystem using "basePath" and the main key? Also, I'd like to move this back into setUp so our cleanup is all in the same place.